src/cart.ts

```diff
@@ -0,0 +1,24 @@
+export type ShipDates = {
+  earliest: Date;
+  latest: Date;
+};
+
+export type PreorderAttributes = {
+  releaseId: string;
+  displayShipDates?: string;
+  estimatedShipDates?: ShipDates; // TODO: Add the raw ship dates to the API
+};
+
+export type CartItem = { variantId: string };
+
+export interface Cart<T extends CartItem> {
+  hasPreorderAttributes: (item: T) => boolean;
```

Comment:
Should this be `getPreorderAttributes: (item: T) => PreorderAttributes | null`?

Reply:
I removed `getPreorderAttributes` because it's actually possible that `addPreorderProperties` is a destructive action in that you can't get back exactly what you put in.

For example if we want this part of the API to deal with formatting the estimated ship date line item property by passing in the actual `Date` instances then it's likely you would lose access to the raw dates when formatted into the property.

If we can figure out a way to use i18next to do the formatting for example maybe we can make this API just deal with the formatted output instead though and then we can probably keep things simple enough for it to always round-trip through add/get.